src/subscription_manager/api.py

```diff
@@ -0,0 +1,83 @@
+#
+# Copyright (c) 2015 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public
+# License as published by the Free Software Foundation; either version
+# 2 of the License (GPLv2) or (at your option) any later version.
+# There is NO WARRANTY for this software, express or implied,
+# including the implied warranties of MERCHANTABILITY,
+# NON-INFRINGEMENT, or FITNESS FOR A PARTICULAR PURPOSE. You should
+# have received a copy of GPLv2 along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+
+"""
+Class and methods in this module are provided for use by external applications with
+the exception of methods beginning with an '_'.
+
+All reasonable efforts will be made to maintain compatibility.
+"""
+import fnmatch
+
+from subscription_manager import logutil
+logutil.init_logger()
+
+from subscription_manager.injectioninit import init_dep_injection
+init_dep_injection()
+
+import subscription_manager.injection as inj
+from subscription_manager.repolib import RepoActionInvoker, RepoFile
+
+
+def _set_enable_for_repositories(setting, *repo_ids):
+    invoker = RepoActionInvoker()
+    repos = invoker.get_repos()
+    repos_to_change = []
+
+    for r in repo_ids:
+        matches = set([repo for repo in repos if fnmatch.fnmatch(repo.id, r)])
+        repos_to_change.extend(matches)
+
+    if len(repos_to_change) == 0:
+        return 0
+
+    # The cache should be primed at this point by the invoker.get_repos()
+    cache = inj.require(inj.OVERRIDE_STATUS_CACHE)
+    identity = inj.require(inj.IDENTITY)
+    cp_provider = inj.require(inj.CP_PROVIDER)
+
+    if identity.is_valid() and cp_provider.get_consumer_auth_cp().supports_resource('content_overrides'):
+        overrides = [{'contentLabel': repo.id, 'name': 'enabled', 'value': setting} for repo in repos_to_change]
+        cp = cp_provider.get_consumer_auth_cp()
+        results = cp.setContentOverrides(identity.uuid, overrides)
+
+        cache = inj.require(inj.OVERRIDE_STATUS_CACHE)
+
+        # Update the cache with the returned JSON
+        cache.server_status = results
+        cache.write_cache()
+
+        invoker.update()
+    else:
+        for repo in repos_to_change:
+            repo['enabled'] = setting
+
+        repo_file = RepoFile()
+        repo_file.read()
+        for repo in repos_to_change:
+            repo_file.update(repo)
+        repo_file.write()
+
+    return len(repos_to_change)
+
+
+def enable_repositories(*repo_ids):
```

Comment:
Or enable_repositories(repo_type='yum', repos) ?

That would theoretically let a 3rd party tool (i'll use 'pip' as an example) change repo info for content added via a subman update_content_hook plugin.). That would mean generalizing the api for setting overrides though (maybe add a update_overrides to ContentActionClient?  So ContentAction classes implement the details, but the
caller of ContentActionClient.update_overrides() could do it in a content type neutral way?)